File: EcommerceProject/EcommerceProject/EcommerceProject/Models/Command.cs
using System;
using System.Collections.Generic;

namespace EcommerceProject
{
    public class Command
    {
        public string id { get; set; }
        public string userId { get; set; }
      
        public List<Product> products { get; set; }
        public double amount { get; set; }
        public string address { get; set; }
        public string status { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

       

        public class Product
        {
            public string id { get; set; }
            public int quantity { get; set; }

            public string productName { get; set; }




            // 
        }
    }
}

File: XamarinProjectEcommerce/src/main/java/com/example/project/models/Command.java
package com.example.project.models;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@Document(collection = "commands")
public class Command {
    @Id
    private String id;
    private String userId;
    private List<Product> products;
    private double amount;
    private String address;
    private String status = "pending";


    @Data
    public static class Product {
        private String productId;
        private String productName;
        private int quantity = 1;
    }
}